pruning/manager.go

```diff
@@ -188,43 +194,54 @@ func (m *Manager) loadPruningSnapshotHeights(db dbm.DB) error {
 	pruneSnapshotHeights := list.New()
 	i, offset := 0, 0
 	for offset < len(bz) {
-		pruneSnapshotHeights.PushBack(int64(binary.BigEndian.Uint64(bz[offset : offset+8])))
+		h := int64(binary.BigEndian.Uint64(bz[offset : offset+8]))
+		if h < 0 {
+			return fmt.Errorf(errNegativeHeightsFmt, h)
+		}
+
+		pruneSnapshotHeights.PushBack(h)
 		i++
 		offset += 8
 	}
 
-	if pruneSnapshotHeights.Len() > 0 {
-		m.mx.Lock()
-		defer m.mx.Unlock()
-		m.pruneSnapshotHeights = pruneSnapshotHeights
-	}
+	m.mx.Lock()
+	defer m.mx.Unlock()
+	m.pruneSnapshotHeights = pruneSnapshotHeights
 
 	return nil
 }
 
 func (m *Manager) flushPruningHeights(batch dbm.Batch) {
-	bz := make([]byte, 0, len(m.pruneHeights) * uint64Size)
-	for _, ph := range m.pruneHeights {
-		buf := make([]byte, uint64Size)
-		binary.BigEndian.PutUint64(buf, uint64(ph))
-		bz = append(bz, buf...)
-	}
-
-	if err := batch.Set([]byte(pruneHeightsKey), bz); err != nil {
+	if err := batch.Set([]byte(pruneHeightsKey), int64SliceToBytes(m.pruneHeights)); err != nil {
 		panic(err)
 	}
 }
 
 func (m *Manager) flushPruningSnapshotHeights(batch dbm.Batch) {
 	m.mx.Lock()
 	defer m.mx.Unlock()
-	bz := make([]byte, 0, m.pruneSnapshotHeights.Len() * uint64Size)
-	for e := m.pruneSnapshotHeights.Front(); e != nil; e = e.Next() {
-		buf := make([]byte, uint64Size)
-		binary.BigEndian.PutUint64(buf, uint64(e.Value.(int64)))
+	if err := batch.Set([]byte(pruneSnapshotHeightsKey), listToBytes(m.pruneSnapshotHeights)); err != nil {
+		panic(err)
+	}
+}
+
+// TODO: convert to a generic version with Go 1.18.
```

Comment:
This this really necessary? How would this change?

If we must do this, let's reference (and create) an issue. I'd like to avoid "naked" TODOs.

Reply:
No, you are right. It's not really necessary unless we start reusing this functionality with other types. I removed the comment